### PR TITLE
Support git.PATH entry in app.ini

### PR DESCRIPTION
**Pull request by typeless**
A user can specify the location of the git executable rather than relying on the PATH environment.

### CODE REVIEW

File: custom/conf/app.ini.sample [hunk 1]
```diff
@@ -636,6 +636,8 @@ SCHEDULE = @every 24h
 UPDATE_EXISTING = true
 
 [git]
+; The path of git executable. If empty, Gitea searches through the PATH environment.
```
Comment (techknowlogick) [inline custom/conf/app.ini.sample:673]: Could you also add this to the cheatsheet in docs/ ?

**Comment by lunny** [comment]
I remember there are some other places use `git` command directly outside of `modules/git`. I think they also should be replaced.

File: custom/conf/app.ini.sample (hunk 1 above)
Comment (typeless) [reply custom/conf/app.ini.sample:673]: Which doc are you referring to? I don't find a file named 'cheatsheet'.

Comment (jolheiser) [reply custom/conf/app.ini.sample:673]: https://github.com/go-gitea/gitea/blob/master/docs/content/doc/advanced/config-cheat-sheet.en-us.md

Comment (techknowlogick) [reply custom/conf/app.ini.sample:673]: Thx for posting whole link :)

**Comment by typeless** [comment]
@lunny I can identify some of them. But I cannot be sure about the effort and time to get them fixed before I look around what is available.

File: custom/conf/app.ini.sample (hunk 1 above)
Comment (typeless) [reply custom/conf/app.ini.sample:673]: Done.

**Comment by lunny** [comment]
@typeless that could be another PR. Also all `process.Manager().ExecDir` should be merged with `git.NewCommand`.

**Comment by lunny** [comment]
conflicted

**Comment by typeless** [comment]
@lunny Done.

I got the error when building it at my local machine. I think it has nothing to do with this change though.
```
 ~/src/code.gitea.io/gitea  ⑂ add-git-path-for-ini    make generate build
go: github.com/denisenkom/go-mssqldb@v0.0.0-20180314172330-6a30f4e59a44: reading https://proxy.golang.org/github.com/denisenkom/go-mssqldb/@v/v0.0.0-20180314172330-6a30f4e59a44.mod: 400 Bad Request      
GO111MODULE=on go generate
go: github.com/denisenkom/go-mssqldb@v0.0.0-20180314172330-6a30f4e59a44: reading https://proxy.golang.org/github.com/denisenkom/go-mssqldb/@v/v0.0.0-20180314172330-6a30f4e59a44.mod: 400 Bad Request      
make: *** [generate] Error 1
```

**Review by techknowlogick — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**